packages/rum-core/src/boot/preStartRum.spec.ts

```diff
@@ -452,6 +502,22 @@ describe('preStartRum', () => {
 
       expect(strategy.initConfiguration).toEqual(initConfiguration)
     })
+
+    it('returns the initConfiguration with the remote configuration when a remoteConfigurationId is provided', (done) => {
```

Comment:
I'm not sure about this behaviour. Basically `DD_RUM.getInitConfiguration()` will also have the remote configuration values. It has the advantage to make the browser extension reflect the remote config without any change.
Wdyt?

Reply:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏ Currently `DD_RUM.getInitConfiguration()` returns exactly what the customer called `init()` with. I wouldn't mind it to return what the SDK was actually initialised with, this means the combination of what the customer called `init()` with, the remote config if applicable and the defaults.

I think it would be useful information to debug rum, both for customers, for us through telemetry and even for support.

I'm not sure if it is safe to do that or if this will be a breaking change.

Reply:
From what I identifyed in [Datadog repos](https://github.com/search?q=org%3ADataDog+getInitConfiguration&type=code), we use the API for
- [Browser SDK extension](https://github.com/DataDog/browser-sdk/blob/b839f8cb52b676241452dba00fa0b8b166cb3bfe/developer-extension/src/panel/hooks/useSdkInfos.ts#L65): to display the conf
- [synthetics-worker](https://github.com/DataDog/synthetics-worker/blob/3735e7b0f1250561272d4527b523ffd58dedcd81/chrome-extension/src/extension/main-world.ts#L71): Use it to get the `applicationId`
- [rum-react-integration-examples](https://github.com/DataDog/rum-react-integration-examples/blob/4fc30362273ac7e126d9767fe67b77a0648ed375/src/Router/RumRouteComponentWrapper.tsx#L58) (old not maintained repo) uses it to get `trackViewsManually`
- [dd-profile-browser](https://github.com/DataDog/dd-profile-browser/blob/e3fd3dcd54ee2970ab74798f3dc7a87571ae458d/src/profiler/rum-configuration-watcher.ts#L19): it is used to collected profiles with `service`, `env`, and `version` tags

Clearly if we get one of fields applicationId, service, env or version, from the RC we will have some issues. But luckily for now they won't be available in RC :).  So the question is, should we anticipate by introducing a new API right away and defining the `getInitConfiguration()` has only returning whats passed to the `init()` API
I'll create an RFC

Reply:
Seen with @BenoitZugmeyer, since the feature is under FF, lets keep the behaviour as is and revisite later when needed.

Reply:
perfect! Sounds good!